src/enclave_proc/resource_manager.rs

```diff
@@ -905,6 +922,168 @@ impl EnclaveManager {
+        measurements: BTreeMap<String, String>,
+    ) -> NitroCliResult<()> {
+        self.enclave_handle
+            .lock()
+            .map_err(|e| {
+                new_nitro_cli_failure!(
+                    &format!("Failed to acquire lock: {:?}", e),
+                    NitroCliErrorEnum::LockAcquireFailure
+                )
+            })?
+            .measurements = measurements;
+        Ok(())
+    }
+
+    /// Read EIF, calculate PCR values and save them.
+    ///
+    /// Go through all the section of the EIF file, extract the headers
+    /// and compute the PCRs using the hashers, based on the section type.
+    /// Measurements will be returned as result of a separate thread executing
+    /// this function.
+    pub fn get_pcrs(eif_path: &str) -> NitroCliResult<BTreeMap<String, String>> {
+        let mut curr_seek = 0;
+        let mut measurements = BTreeMap::new();
+        let mut eif_file = File::open(eif_path).map_err(|e| {
+            new_nitro_cli_failure!(
+                &format!("Failed to open the EIF file: {:?}", e),
+                NitroCliErrorEnum::FileOperationFailure
+            )
+            .add_info(vec![eif_path, "Open"])
+        })?;
+
+        // Skip EIF header
+        curr_seek += EifHeader::size();
+        eif_file
+            .seek(SeekFrom::Start(curr_seek as u64))
+            .expect("Could not seek kernel to first section.");
+
+        let mut section_buf = vec![0u8; EifSectionHeader::size()];
+        let mut image_hasher =
+            EifHasher::new_without_cache(Sha384::new()).expect("Could not create image_hasher");
+        let mut bootstrap_hasher =
+            EifHasher::new_without_cache(Sha384::new()).expect("Could not create bootstrap_hasher");
+        let mut app_hasher =
+            EifHasher::new_without_cache(Sha384::new()).expect("Could not create app_hasher");
+        let mut cert_hasher =
+            EifHasher::new_without_cache(Sha384::new()).expect("Could not create cert_hasher");
+        let mut ramdisk_idx = 0;
+
+        // Read all section headers and treat by type
+        while eif_file
+            .read_exact(&mut section_buf)
+            .map_err(|e| {
+                new_nitro_cli_failure!(
+                    &format!("Error while reading EIF header: {:?}", e),
+                    NitroCliErrorEnum::EifParsingError
+                )
+            })
+            .is_ok()
+        {
+            let section = EifSectionHeader::from_be_bytes(&section_buf).map_err(|e| {
+                new_nitro_cli_failure!(
+                    &format!("Error extracting EIF section header: {:?}", e),
+                    NitroCliErrorEnum::EifParsingError
+                )
+            })?;
+
+            let mut buf = vec![0u8; section.section_size as usize];
+            curr_seek += EifSectionHeader::size();
+            eif_file
+                .seek(SeekFrom::Start(curr_seek as u64))
+                .expect("Could not seek EIF after header");
```

Comment:
Expect exits the process with the mentioned error. Nitro Enclaves CLI has it's own failure printing through new_nitro_cli_failure! macro. It would be better to use this custom failure printing as much as possible, this way all errors will look the same. This comment is referring to all appearances of expect in your code.

Reply:
Handled in the following commit after the requested refactoring by @alexggh